C104 make it restful

HadarRose opened the pull request:
Implemented Dropwizard:

- Moved code from postTweet and getTimeline classes to DropwizardResource
- Changed packaging plugin from assembly to shade
- Added necessary Dropwizard classes and packages
- Updated README

Commit:
Hello World Dropwizard classes added

Added multiple classes and resources to work with theDropwizard hello world example. Edited pom.xml for same reasons. Will most likely require a lot of editing and refactoring

Commit:
Draft

very unfinished product pushed to share my progress and not for actual production

Commit:
Implemented Dropwizard

-deleted old java classes for tweeting and timeline and added them to DropwizardResource
- Created new classes for Dropwizard setup
- changed maven package plugin from assemply to shade
- updated readme

kev-chan commented on src/main/java/resources/DropwizardResource.java:
```diff
@@ -0,0 +1,59 @@
+        }
+    }
+
+    public static class Message{
```
dont use static or inner classes

kev-chan commented on src/test/tests/PostTweetTest.java:
```diff
@@ -21,20 +21,20 @@
 //    public void noArgs() throws TwitterException {
 //        String[] noArgs = {};
-//        PostTweet.main(noArgs);
+//        resources.PostTweet.main(noArgs);
```
PostTweet classes is deleted.  Do you still need this?

kev-chan commented on src/main/java/resources/DropwizardResource.java:
```diff
@@ -0,0 +1,59 @@
+
+@Path("/api/1.0/twitter")
+@Produces(MediaType.APPLICATION_JSON)
+public class DropwizardResource {
```
how did you decide on this className and is this the best name for the class?

kev-chan commented:
ensure all errors come across the same way

Commit:
Error Handling altered

Changed error handling to go through a new class: ExceptionHandler. Made all error messages be in the same format. Added to .gitignore

Commit:
Added more specific error messages

kev-chan commented on src/main/java/DropwizardApplication.java:
```diff
@@ -0,0 +1,22 @@
+import io.dropwizard.setup.Environment;
+import resources.DropwizardResource;
+
+public class DropwizardApplication extends Application<DropwizardConfig>{
```
is this the best name for this class?

Commit:
Added more specific error messages

Commit:
Refactoring and clarifying unauthorized error message

kev-chan commented on src/main/java/Runner.java:
```diff
@@ -0,0 +1,22 @@
+import io.dropwizard.setup.Environment;
+import resources.TwitterRequestHandler;
+
+public class Runner extends Application<InputConfiguration>{
```
Runner is pretty generic

kev-chan commented on src/main/java/api/ErrorMessage.java:
```diff
@@ -0,0 +1,26 @@
+
+
+public class ErrorMessage {
+    private int statusCode;
```
whitespace should be consistent

kev-chan commented on src/main/java/resources/TwitterRequestHandler.java:
```diff
@@ -0,0 +1,55 @@
+            return Response.ok(new StatusList(statusList)).build();
+        } catch (Exception e){
+            LOGGER.error(e.getMessage(), e); // log error
+            return ExceptionHandler.ExceptionHandler(e);
```
that's a pretty odd name for a method, I'm sure it can be better.  Method name should describe what they are doing.

kev-chan commented on src/main/java/resources/TwitterRequestHandler.java:
```diff
@@ -0,0 +1,55 @@
+
+    @POST
+    @Path("/tweet")
+    public Response postTweet(Message post){
```
you are writing an api so consider what other engineers/developers may use your api for.  This should impact what is returned.  Please take a look at what you have.

HadarRose replied:
Should these calls only return status codes and not entire responses?

kev-chan replied:
you got there

kev-chan commented on src/main/java/InputConfiguration.java:
```diff
@@ -0,0 +1,7 @@
+// but in reality, we don't have anything to configure atm
+import io.dropwizard.Configuration;
+
+public class InputConfiguration extends Configuration{
```
InputConfiguration sounds like it is data entered in

Commit:
Refactored: renamed several classes and methods

Commit:
Change Post  output to status instead of text

kev-chan commented on src/main/java/TwitterApplication.java:
```diff
@@ -0,0 +1,22 @@
+import io.dropwizard.setup.Environment;
+import resources.TwitterRequestHandler;
+
+public class TwitterApplication extends Application<ApplicationConfiguration>{
```
much better name.  I normally wouldn't call it twitter app since it could expand beyond twitter but at least it's informative!

kev-chan commented on src/main/java/resources/TwitterRequestHandler.java:
```diff
@@ -0,0 +1,54 @@
+
+@Path("/api/1.0/twitter")
+@Produces(MediaType.APPLICATION_JSON)
+public class TwitterRequestHandler {
```
call it  <Blank>Resource to be consistent with dropwizard verbiage

kev-chan commented on src/main/java/resources/TwitterRequestHandler.java:
```diff
@@ -0,0 +1,54 @@
+    @GET
+    @Path("/timeline")
+    public Response getTimeline(){
+        LOGGER.info("GET request at /api/1.0/twitter/timeline was triggered");
```
ideally you don't explicitly log the version in plaintext as the route could change and this message would no longer be accurate

kev-chan reviewed (changes requested):
also the indention

Commit:
code beautified

Commit:
Refactored resource class, removed version hardcoding from logs

Commit:
Credentials error changed to general

kev-chan reviewed (approved):
the Message should not be in api package but you can resolve that in next lab
